flink-runtime/src/main/java/org/apache/flink/runtime/operators/resettable/AbstractBlockResettableIterator.java

```diff
@@ -183,4 +183,24 @@ protected T getNextRecord() throws IOException {
 		}
 	}
 
+	public void resetForIterativeTasks() {
```

Comment:
The already existing 'reset' tries to reset with in the logic of iteration.
`this.readView.setReadPosition(0);
		this.numRecordsReturned = 0;`

But it does not put back the fullSegments back to the emptySegment. We need a much enhanced reset version. Hence I did not want to change the available one and hence added a new one.  WDYT?